web/access_control.go

```diff
@@ -108,3 +109,16 @@ func (s *AutograderService) getUserAndSCM2(ctx context.Context, courseID uint64,
 	}
 	return s.getUserAndSCM(ctx, crs.GetProvider(), mustBeAdmin)
 }
+
+// teacherScopes defines scopes that must be enabled for a teacher token to be valid.
+var teacherScopes = []string{"admin:org", "delete_repo", "repo", "user"}
+
+// hasTeacherScopes checks whether current user has upgraded scopes on provided scm client.
+func (s *AutograderService) hasTeacherScopes(ctx context.Context, sc scm.SCM) bool {
+	auth := sc.GetUserScopes(ctx)
+	if !cmp.Equal(auth.Scopes, teacherScopes) {
```

Comment:
@0xf8f8ff This `cmp.Equal` I suppose is sensitive to the specific order of the entries coming from github and so, if the order changes in the future, this may fail, or?

Reply:
Yes, it is order sensitive. I'll add sort before compare, probably as a separate function.